commit 7b6e72e53a01edafb0bc10fdf51e58f7f7f88831 (refs/changes/19/5119/1)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-23T10:27:54-08:00 (11 months ago)
    
    TRITON-987 fabric setup broken by needless dependency on broken CA

diff --git a/CHANGES.md b/CHANGES.md
index b269f56..16b97b7 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.22.0
+
+- TRITON-987 fabric setup broken by needless dependency on broken CA
+
 ## 1.21.6
 
 - TOOLS-2114 sdcadm treats server as "unavailable" when it doesn't have a transitional_status
diff --git a/lib/cli/do_update_other.js b/lib/cli/do_update_other.js
index b370036..16e74bd 100644
--- a/lib/cli/do_update_other.js
+++ b/lib/cli/do_update_other.js
@@ -22,6 +22,14 @@ var common = require('../common');
 var errors = require('../errors');
 var steps = require('../steps');
 
+
+/*
+ * These services are broken and have been removed from new installs and the
+ * agentsshar. We want to skip them here so we're not broken by them being
+ * missing.
+ */
+var SERVICE_BLACKLIST = ['cabase', 'cainstsvc'];
+
 /*
  * These services haven't always existed and don't have scripts in
  * sdc-headnode that create SDC app metadata in SAPI for them. We
@@ -146,7 +154,11 @@ function do_update_other(subcmd, opts, args, cb) {
                     return;
                 }
 
-                ctx.svcs = svcs;
+                // Filter out any blacklisted services
+                ctx.svcs = svcs.filter(function ignoreBlacklistedServices(svc) {
+                    return (SERVICE_BLACKLIST.indexOf(svc.name) === -1);
+                });
+
                 ctx.svcFromName = {};
                 svcs.forEach(function (svc) {
                     ctx.svcFromName[svc.name] = svc;
diff --git a/lib/steps/sapi.js b/lib/steps/sapi.js
index b6000d9..abd481b 100644
--- a/lib/steps/sapi.js
+++ b/lib/steps/sapi.js
@@ -52,8 +52,6 @@ function ensureAgentServices(arg, cb) {
         'agents_core',
         'amon-agent',
         'amon-relay',
-        'cabase',
-        'cainstsvc',
         'cmon-agent',
         'cn-agent',
         'config-agent',
@@ -230,7 +228,6 @@ function ensureAgentServices(arg, cb) {
                         callback();
                         return;
                     }
-                    progress('Updating service for agent \'%s\'', agent);
                     log.trace({
                         service: agent,
                         params: agentServices[agent]
diff --git a/package.json b/package.json
index 3980180..29eab86 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.21.6",
+  "version": "1.22.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
